olive/passes/onnx/nvmo_quantization.py

```diff
@@ -54,46 +51,360 @@ def _default_config(cls, accelerator_spec: AcceleratorSpec) -> Dict[str, PassCon
+                add_position_ids,
+            )
+            inputs = {input_name: torch_tensor.cpu().numpy() for input_name, torch_tensor in inputs.items()}
+            batched_inputs_list.append(inputs)
+
+        return batched_inputs_list
+
+    def quantize_awq(self, model: Union[ModelProto, str], quant_config: Dict[str, Any]) -> ModelProto:
+        """Perform nvidia_awq quantization using ModelOpt's int4 quantize function.
+
+        Args:
+            model (ModelProto | str): The ONNX model or path to the model to quantize.
+            quant_config (Dict[str, Any]): Configuration dictionary for quantization.
+
+        Returns:
+            ModelProto: The quantized ONNX model.
+
+        """
+        try:
+            from modelopt.onnx.quantization.int4 import quantize as quantize_int4
+        except ImportError:
+            logger.exception(
+                "Please ensure that 'modelopt' package is installed. Install it with 'pip install nvidia_modelopt'."
+            )
+            raise ImportError(
+                "modelopt is not installed. Please install it using 'pip install nvidia_modelopt'. Exiting."
+            ) from None
+
+        logger.debug("Starting nvidia_awq quantization...")
+
+        # Prepare calibration inputs
+        calib_inputs = quant_config["calibration_data_reader"]
+
+        # Perform quantization using ModelOpt's int4 quantize function
+        quantized_model = quantize_int4(
+            model,
+            calibration_method=quant_config["calibration_method"],
+            calibration_data_reader=calib_inputs,
+        )
+
+        logger.debug("Completed nvidia_awq quantization.")
+        return quantized_model
+
+    def convert_opset_to_21_proto(self, model_proto: ModelProto) -> ModelProto:
+        """Modify the model's opset to 21 if it's not already, operating on a ModelProto.
+
+        Args:
+            model_proto (ModelProto): The ONNX model proto to modify.
+
+        Returns:
+            ModelProto: The updated ONNX model proto with opset version 21.
+
+        """
+        current_opset = {opset.domain: opset.version for opset in model_proto.opset_import}
+
+        default_domain_version = current_opset.get("", 0)
+        if default_domain_version >= 21:
+            logger.debug(
+                "Model already uses opset version %s for the default domain. Skip conversion.", default_domain_version
+            )
+            return model_proto  # No conversion needed
+
+        new_opset_imports = [
+            helper.make_opsetid("", 21),  # Default domain with opset version 21
+            helper.make_opsetid("com.microsoft", 1),  # Microsoft domain with version 1
+        ]
+
+        for domain, version in current_opset.items():
+            if domain not in ["", "com.microsoft"]:
+                new_opset_imports.append(helper.make_opsetid(domain, version))
+
+        # Update the model's opset imports
+        model_proto.opset_import.clear()
```

Comment:
I am able to replicate the pipeline issue on my Linux machine with onnx 1.17.0 and protobuf 3.20.2/3.20.3. I am not sure what the cause of the issue is, but it works fine if you replace this line with `model_proto.ClearField("opset_import")`. 
Maybe onnx is picking up an older version of protobuf in our systems.

Could you update this line?

Reply:
Yes, using model_proto.ClearField("opset_import") works at my end also. Validate the unit test and phi3 example. Both are working fine

Reply:
I have updated the commit, could you please trigger the pipeline